homeassistant/components/hive/climate.py

```diff
@@ -42,6 +55,19 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     add_entities([climate])
 
+    def heating_boost(service):
```

Comment:
i didnt want to add the service handlers and registration to the componet as the services should not registered if the user does not have heating or hot water. E.G they only had hive lights and a hub, the user should see a heating or water heater service.

Reply:
I think we should sort the node ids per ha_type so that we only load platform once per ha_type. Then we can also register the corresponding service,from the component, when iterating the sorted dict. Probably use a dict with lists of devices.

Reply:
i believe thats what we currently have now we just drill down and initialise each device in the component instead of the platform. how would you suggest we did it change the load platform to itertate through each device adding an entity in the setup platform function?

DEVICETYPES = {
    "binary_sensor": "device_list_binary_sensor",
    "climate": "device_list_climate",
    "water_heater": "device_list_water_heater",
    "light": "device_list_light",
    "switch": "device_list_plug",
    "sensor": "device_list_sensor",
}

Reply:
Something like this but simplified:
https://github.com/home-assistant/home-assistant/blob/5914475fe5f97d08e9d721cbf328727df2f3af5e/homeassistant/components/mysensors/gateway.py#L186-L196

Reply:
Pass the list of devices to `load_platform`. In each platform, iterate this list to create all the entities to add.